### PR TITLE
K8SPSMDB-391: Remove RS from status after shard was removed

### PR DESCRIPTION
**Pull request by blez**
[K8SPSMDB-391](https://jira.percona.com/browse/K8SPSMDB-391) [&#10088;?&#10089;](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

### CODE REVIEW

File: pkg/controller/perconaservermongodb/status.go [hunk 1]
```diff
@@ -66,6 +66,17 @@ func (r *ReconcilePerconaServerMongoDB) updateStatus(cr *api.PerconaServerMongoD
 		}
 	}
 
+	if len(repls) < len(cr.Status.Replsets) {
```
Comment (delgod) [inline pkg/controller/perconaservermongodb/status.go:69]: what if `len` is the same? I mean, the user "renamed" shard in CR

**Review by delgod — CHANGES_REQUESTED** [review]

File: pkg/controller/perconaservermongodb/status.go (hunk 1 above)
Comment (blez) [reply pkg/controller/perconaservermongodb/status.go]: https://github.com/percona/percona-server-mongodb-operator/pull/556/commits/00758a20f7e7223e6f23af1eb446b4ba443e9e7b

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| security-context | passed |
| demand-backup-sharded | passed |
| rs-shard-migration | passed |
| upgrade | passed |
| upgrade-sharded | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| security-context | passed |
| operator-self-healing | passed |
| demand-backup-sharded | passed |
| rs-shard-migration | passed |
| upgrade | passed |
| upgrade-sharded | passed |

**Review by delgod — APPROVED** [review]

**Review by nonemax — APPROVED** [review]

**Comment by JNKPercona** [comment]
PSMDB operator docker - perconalab/percona-server-mongodb-operator:PR-556-00758a20

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| storage | passed |
| security-context | passed |
| self-healing | passed |
| rs-shard-migration | passed |
| operator-self-healing | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |

**Pull request merged**